### PR TITLE
PP-92 Fixing get contract for merge addresses

### PR DESCRIPTION
**Pull request by wilsoniovlabs**
## What

- Review the SDK to find issues and fixed

## Why

- New error added when fix the contract addresses
- Review the SDK to find error to initialized (The error is to get all contracts from rif-relay-contracts json contract list file.

## Refs

- (optional: include links to other issues, PRs, tickets, etc)

### CODE REVIEW

**Review by mortelli — APPROVED** [review]
LGTM :+1: 

i would verify 2 things here:
1. see if the error bubbles up to the user in the dapp
2. verify `dist` is up-to-date

**Comment by mortelli** [comment]
also @wilsoniovlabs: any particular reason why the package lock file is being included here?

File: src/contracts.ts [hunk 1]
```diff
@@ -24,6 +24,11 @@ export class Contracts {
         contractAddresses = contractAddresses ?? <RelayingServicesAddresses>{};
         const contracts: RelayingServicesAddresses =
             getContractAddresses(chainId);
+        if (!contracts) {
```
Comment (antomor) [inline src/contracts.ts:27]: I think that raising an error here won't allow the user to initialize the `Contracts` object.
We could move the error (maybe creating a custom error) in the `getContractAddresses` and catch the error here (maybe adding some warnings) to be able to complete the `Contracts` initialization.

**Review by antomor — CHANGES_REQUESTED** [review]
@wilsoniovlabs thanks for the changes.
Apart from the comment I left, it's not clear to me if the lib should be built whether with `npm run build` or with `npx webpack`. I cannot find the related section in the readme, so I would add some instructions about that. However, if one of the two methods is not required anymore, I would remove it.

File: src/ContractError.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export class ContractError extends Error {
+    constructor(msg: string) {
+        super(msg);
```
Comment (antomor) [inline src/ContractError.ts:3]:
```suggestion
        super(`ContractError: ${msg}`);
```

File: src/ContractError.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+        Object.setPrototypeOf(this, ContractError.prototype);
```
Comment (antomor) [inline src/ContractError.ts:5]: I like it, would you mind adding a reference to the reason behind this line?
```suggestion
        // https://github.com/Microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work
        Object.setPrototypeOf(this, ContractError.prototype);
```

File: src/utils.ts [hunk 4]
```diff
@@ -31,7 +32,13 @@ export function getContract(
 export function getContractAddresses(
     chainId: number
 ): RelayingServicesAddresses {
-    return ContractAddresses[chainId];
+    const contracts = ContractAddresses[chainId];
+    if (!contracts) {
+        throw new ContractError(
+            `ContractError: contracts not find for the specified network id ${chainId}.`
```
Comment (antomor) [inline src/utils.ts:38]: If you accept the suggestion on the ContractError constructor:
```suggestion
            `No contracts found for the specified network id ${chainId}.`
```

**Review by antomor — CHANGES_REQUESTED** [review]
LGTM, Thank you @wilsoniovlabs! Just some minor changes before the approval.

**Review by antomor — APPROVED** [review]

**Pull request merged**